File: growing-shop-java/src/main/java/com/example/growingshop/domain/user/dto/UserResponse.java
package com.example.growingshop.domain.user.dto;

import com.example.growingshop.domain.company.dto.CompanyResponse;
import com.example.growingshop.domain.user.domain.User;
import com.example.growingshop.domain.user.domain.UserGrade;
import com.example.growingshop.domain.user.domain.UserType;
import lombok.*;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class UserResponse {

    @NoArgsConstructor(access = AccessLevel.PRIVATE)
    @AllArgsConstructor
    @Builder
    @Getter
    public static class UserRes {
        private Long id;
        private String name;
        private String mobile;
        private String email;
        private String loginId;
        private CompanyResponse.CompanyRes company;
        private UserType type;
        private UserGrade grade;

        public static UserRes from(User user) {
            UserRes res = new UserRes();

            res.id = user.getId();
            res.name = user.getName();
            res.mobile = user.getMobile();
            res.email = user.getEmail();
            res.loginId = user.getLoginId();

            if (user.getCompany() != null) {
                res.company = CompanyResponse.CompanyRes.from(user.getCompany());
            }

            res.type = user.getType();
            res.grade = user.getGrade();

            return res;
        }
    }
}

File: growing-shop-api/auth/src/main/kotlin/com/example/growingshopauth/user/dto/UserResponse.kt
package com.example.growingshopauth.user.dto

import com.example.domain.user.User
import com.example.domain.user.UserGrade
import com.example.domain.user.UserType
import com.example.growingshopauth.company.dto.CompanyResponse

class UserResponse {

    class UserRes(
        val id: Long,
        val name: String,
        val mobile: String,
        val loginId: String,
        val email: String,
        val company: CompanyResponse.CompanyRes?,
        val type: UserType,
        val grade: UserGrade
    ) {
        companion object {
            fun from(user: User): UserRes {
                return UserRes(
                    user.id!!,
                    user.name,
                    user.mobile,
                    user.email,
                    user.loginId,
                    user.company?.let { CompanyResponse.CompanyRes.from(it) },
                    user.type,
                    user.grade
                )
            }
        }
    }
}
